src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -260,7 +261,6 @@ ClassFileStream* ClassPathDirEntry::open_stream(JavaThread* current, const char*
         if (UsePerfData) {
           ClassLoader::perf_sys_classfile_bytes_read()->inc(num_read);
         }
-        FREE_RESOURCE_ARRAY(char, path, path_len);
```

Comment:
I see your argument about this being basically a no-op, but we also lose the guard against access-after-free (ZapResourceArea). At a minimum I'd add a comment "// No need to free path here" else someone will think it is an oversight.

Reply:
`ZapResourceArea` is only enabled when ASSERT is defined, so I bring back FREE_RESOURCE_ARRAY_IN_THREAD under ASSERT. We won't call Afree() in release build.